doc/conf.py

```diff
@@ -45,10 +47,29 @@
     "README.md",
 ]
 
+# -- Options for templating --------------------------------------------------
+
+templates_path = ["_templates"]
+
 # -- Options for HTML output -------------------------------------------------
 
 html_theme = "sphinx_rtd_theme"
 
+git_describe = ["git", "describe", "--tags", "--exact-match"]
+git_symref = ["git", "symbolic-ref", "--short", "HEAD"]
+
+try:
+    git_version = check_output(git_describe, stderr=DEVNULL, encoding="utf-8")
+except CalledProcessError:
+    git_version = check_output(git_symref, stderr=DEVNULL, encoding="utf-8")
+
+html_context = {
+    "VERSIONS": True,
+    "current_version": git_version.strip(),
+}
+html_js_files = ["versions.js"]
+html_static_path = ["_static"]
```

Comment:
It might be helpful to leave comments that state which files make use of this information. (For instance, it appears that `current_version` is used by `doc/_templates/versions.html`.)

Reply:
Clarified in efd39c40cd6ad68aee4d384cbb489c7707904d0d